dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/RefseqToGenbankMappingProcessorTest.java

```diff
@@ -0,0 +1,91 @@
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import uk.ac.ebi.eva.dbsnpimporter.contig.ContigMapping;
+import uk.ac.ebi.eva.dbsnpimporter.models.LocusType;
+import uk.ac.ebi.eva.dbsnpimporter.models.Orientation;
+import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class RefseqToGenbankMappingProcessorTest {
+
+    private static final String REFSEQ_CONTIG = "NT_example";
+
+    private static final String EQUIVALENT_GENBANK_CONTIG = "GK_example";
+
+    private static final String CHROMOSOME = "4";
+
+    private RefseqToGenbankMappingProcessor refseqToGenbankMappingProcessor;
+
+    @Before
+    public void setUp() throws Exception {
+        Map<String, String> contigMap = new HashMap<>();
+        contigMap.put(REFSEQ_CONTIG, EQUIVALENT_GENBANK_CONTIG);
+        ContigMapping contigMapping = new ContigMapping(contigMap);
+        refseqToGenbankMappingProcessor = new RefseqToGenbankMappingProcessor(contigMapping);
+    }
+
+    @Test
+    public void replaceContigIfChromosomeRegionIsNotValid() throws Exception {
+        SubSnpCoreFields subSnpCoreFields = new SubSnpCoreFields(1L, Orientation.FORWARD, 1L, Orientation.FORWARD,
+                                                                 REFSEQ_CONTIG, 1L, 1L, Orientation.FORWARD,
+                                                                 LocusType.SNP, "4", null, null, "T", "T", "A", "T/A",
+                                                                 "",
+                                                                 null, null, Orientation.FORWARD, null, null, null,
+                                                                 Orientation.FORWARD, "batch");
+
+        SubSnpCoreFields processed = refseqToGenbankMappingProcessor.process(subSnpCoreFields);
+
+        assertEquals(EQUIVALENT_GENBANK_CONTIG, processed.getVariantCoreFields().getChromosome());
+    }
+
+    @Test
+    public void dontReplaceContigIfChromosomeRegionIsValid() throws Exception {
+        SubSnpCoreFields subSnpCoreFields = new SubSnpCoreFields(1L, Orientation.FORWARD, 1L, Orientation.FORWARD,
+                                                                 REFSEQ_CONTIG, 1L, 1L, Orientation.FORWARD,
+                                                                 LocusType.SNP, CHROMOSOME, 1L, 1L, "T", "T", "A",
+                                                                 "T/A", "",
+                                                                 null, null, Orientation.FORWARD, null, null, null,
+                                                                 Orientation.FORWARD, "batch");
+
+        SubSnpCoreFields processed = refseqToGenbankMappingProcessor.process(subSnpCoreFields);
+
+        assertEquals(CHROMOSOME, processed.getVariantCoreFields().getChromosome());
```

Comment:
This test would be more complete adding this:
assertEquals(REFSEQ_CONTIG, processed.getContigRegion().getChromosome());

Anyway, it does not affect how we are using the processed object (we don't call getContigRegion() ). In the other hand, that method is public.